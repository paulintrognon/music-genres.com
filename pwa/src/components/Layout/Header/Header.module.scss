@import '../../../styles/mixins.scss';

.container {
  max-width: var(--max-width);
  margin: auto;
  padding: 2rem 1rem;


  @include breakpoint(21rem) {
    display: flex;
    justify-content: space-between;
    align-items: top;
  }
}

.logo img {
  @include breakpoint(0, phone) {
    width: 6rem;
  }
}

.sideButton {
  border: none;
  text-transform: uppercase;
  font-size: var(--font-size-14);
  font-weight: 700;

  &:not(:first-child) {
    margin-left: 2rem;
  }
}
