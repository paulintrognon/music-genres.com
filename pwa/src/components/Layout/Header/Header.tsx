import Link from 'next/link'
import classes from './Header.module.scss'

const Header: React.FC = () => {
  return (
    <header className={classes.container}>
      <h1 className={classes.logo}>
        <Link href="/">
          <a>
            <img src="/images/logo.png" alt="Music Genres" />
          </a>
        </Link>
      </h1>
      <div className={classes.sideButtons}>
        <Link href="/all">
          <a className={classes.sideButton}>All genres</a>
        </Link>
        <Link href="/add/video">
          <a className={classes.sideButton}>Add genre</a>
        </Link>
      </div>
    </header>
  )
}
export default Header
