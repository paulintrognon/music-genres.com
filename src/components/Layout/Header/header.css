.main-header {
  padding: 30px 40px;
}

.main-header .main-logo {
  display: inline-block;
}

.main-header .side-buttons {
  float: right;
  margin-top: 24px;
}

.main-header .side-buttons .side-button {
  border: 2px solid white;
  padding: 10px 22px;
  margin-left: 20px;
}

.main-header .side-buttons .side-button img {
  margin-left: 10px;
  vertical-align: middle;
}
