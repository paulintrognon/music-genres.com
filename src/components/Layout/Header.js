import React from 'react';
import { Link } from 'react-router-dom';

import './header.css';

export default function () {
  return (
    <header className="main-header">
      <nav className="main-nav">
        <Link to={'/'} className="main-nav-brand">music-genres.com</Link>

        <div className="side-links">
          <Link to={'/browse'} className="side-link">Browse</Link>
          <Link to={'/random'} className="side-link">Random</Link>
        </div>
      </nav>
    </header>
  )
}
