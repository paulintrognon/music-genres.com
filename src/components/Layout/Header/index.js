import React from 'react';
import { Link } from 'react-router-dom';

import './header.css';
import logo from './logo.png';
import Plus from '../../images/Plus';

export default class Header extends React.Component {
  render() {
    return (
      <header className="main-header">
        <div className="main-logo">
          <Link to='/'>
            <img src={logo} alt="Music Genres"/>
          </Link>
        </div>
        <div className="side-buttons">
          <Link to={'/tracks/add'} className="side-button">
            Add video
            <Plus></Plus>
          </Link>
        </div>
      </header>
    );
  }
}
