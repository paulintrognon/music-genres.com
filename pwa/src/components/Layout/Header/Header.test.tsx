import Header from './Header'
import { shallow } from 'enzyme'

it('should render the Header', () => {
  const wrapper = shallow(<Header />)
  expect(wrapper).toMatchSnapshot()
})
