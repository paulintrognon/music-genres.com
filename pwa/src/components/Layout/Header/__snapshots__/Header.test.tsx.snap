// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Header 1`] = `
<header
  className="container"
>
  <h1
    className="logo"
  >
    <Link
      href="/"
    >
      <a>
        <img
          alt="Music Genres"
          src="/images/logo.png"
        />
      </a>
    </Link>
  </h1>
  <div
    className="sideButtons"
  >
    <Link
      href="/all"
    >
      <a
        className="sideButton"
      >
        All genres
      </a>
    </Link>
    <Link
      href="/add/video"
    >
      <a
        className="sideButton"
      >
        Add genre
      </a>
    </Link>
  </div>
</header>
`;
